src/codemagic/models/xcodebuild.py

```diff
@@ -190,12 +206,18 @@ def clean(self):
         cmd = [*self._construct_base_command(None), 'clean']
         self._run_command(cmd, f'Failed to clean {self.workspace or self.project}')
 
-    def archive(self,
-                export_options: ExportOptions,
-                archive_directory: pathlib.Path,
-                *,
-                xcargs: Optional[str] = None,
-                custom_flags: Optional[str] = None) -> pathlib.Path:
+    def archive(
+        self,
+        export_options: ExportOptions,
+        archive_directory: pathlib.Path,
+        *,
+        xcargs: Optional[str] = None,
+        custom_flags: Optional[str] = None,
+        xcode: Optional[Xcode] = None,
+    ) -> pathlib.Path:
+        if xcode is None:
```

Comment:
Could you elaborate on this?

We only call archive in one place, where we pass `xcode` as a parameter. And there, we get the `xcode` version by the same `Xcode.get_selected()` call, which, if all is well, should return the version - though in case of error, it could also return `None` - but I don't think we even reach this then. And even if we did, it seems weird that we rely on the same method, in that case, moments later.

I would say either this argument is not optional, or we could, in fact, not even attempt to pass it (and resolve it within the `archive` call). Currently, it just looks like double trouble. Am I missing something?